File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0) return false;
        if(x<10) return true;
        int num=0;
        int y=x;
        while(x>0){
            num=num*10+x%10;
            x/=10;
        }
        return y==num;
    }
}

File: 9-palindrome-number/palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if(x<0) return false;
    const str=x+'';
    const n=str.length;
    let i=0;
    let j=n-1;
    while(i<n && j<n){
        if(str[i]!==str[j]) return false;
        i++;
        j--;
    }
    return true;
};